plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/date/DateParserFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to ElasticSearch and Shay Banon under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. ElasticSearch licenses this
+ * file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.date;
+
+import org.joda.time.DateTimeZone;
+
+import java.util.Locale;
+
+public class DateParserFactory {
+
+    public static final String ISO8601 = "ISO8601";
+    public static final String UNIX = "UNIX";
+    public static final String UNIX_MS = "UNIX_MS";
+    public static final String TAI64N = "TAI64N";
+
+    public static DateParser createDateParser(String format, DateTimeZone timezone, Locale locale) {
+        switch(format) {
+            case ISO8601:
+                // TODO(talevy): fallback solution for almost ISO8601
```

Comment:
This was a placeholder to remind us that `logstash-filter-date` accepts more lenient forms of ISO8601 that may not match exactly. I am not sure how important this is. Maybe @jordansissel can shed some light on why these more lenient forms came to be.

here is the logstash code:
[file: date.rb](https://github.com/logstash-plugins/logstash-filter-date/blob/master/lib/logstash/filters/date.rb#L142-L149)
```ruby
...
#Fall back solution of almost ISO8601 date-time
almostISOparsers = [
org.joda.time.format.DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSSZ").getParser(),
org.joda.time.format.DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS").getParser(),
org.joda.time.format.DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss,SSSZ").getParser(),
org.joda.time.format.DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss,SSS").getParser()
].to_java(org.joda.time.format.DateTimeParser)
...
```